release.nix

```diff
@@ -58,25 +58,20 @@ let
   } // (mkRequiredJob (
       collectTests jobs.native.checks.tests ++
       collectTests jobs.native.benchmarks ++
-      [ jobs.native.byronLedgerSpec.x86_64-linux
-        jobs.native.byronChainSpec.x86_64-linux
-        jobs.native.semanticsSpec.x86_64-linux
-        jobs.native.shelleyLedgerSpec.x86_64-linux
-        jobs.native.delegationDesignSpec.x86_64-linux
-        jobs.native.nonIntegerCalculations.x86_64-linux
-        jobs.native.blocksCDDLSpec.x86_64-linux
-      ]
+      mapAttrsToList (_: spec: spec.x86_64-linux or null) jobs.native.specs
     ))
 
-  # Collect all spec PDFs, without system suffix
-  // { inherit (project)
-         byronLedgerSpec
-         byronChainSpec
-         semanticsSpec
-         shelleyLedgerSpec
-         delegationDesignSpec
-         nonIntegerCalculations
-         blocksCDDLSpec
-       ; };
+  // {
+    # Collect all spec PDFs, without system suffix.
+    specs = removeAttrs project.specs ["recurseForDerivations"];
```

Comment:
Could you explain a bit more what's going on here?

Reply:
Sorry, too late to update the code comment, but... this is related to how nix and hydra decide whether they should descend into attrsets to search for more derivations.

This `removeAttrs` will undo the effects of `pkgs.recurseIntoAttrs`. If the `{ recurseForDerivations = true; }` attr remains, Hydra will make an eval error saying a boolean is not a job. So the boolean attr must be removed. This isn't required if the attrset is passed the "normal way" through `mapTestOn` and `packagePlatforms`.